cassiopeia/simulator/DataSimulator.py

```diff
@@ -0,0 +1,19 @@
+"""
+Abstract class DataSimulator, for overlaying data onto a CassiopeiaTree.
+
+All data simulators are derived classes of this abstract class, and at a minimum
+implement a method called `overlay_data`.
+"""
+import abc
+from cassiopeia.data import CassiopeiaTree
+
+
+class DataSimulator(abc.ABC):
+    """
+    DataSimulator is an abstract class that all data overlayers derive from.
+    """
+
+    @abc.abstractmethod
+    def overlay_data(self, tree: CassiopeiaTree) -> None:
+        """Overlay data on a CassiopeiaTree (in-place)"""
```

Comment:
can you document this function a bit more? What do we expect to be in `tree`, what do we not expect? Will this modify the tree in place? etc.

Reply:
The docstring already specifies that the tree is modified in place (perhaps it wasn't clear?). I can't think about any more specific documentation that this at the moment, since baed on our discussion I believe that `DataSimulator` is a very generic class. Maybe we should remove it completely until we find a need for it? (right now only `LineageTracingDataSimulator` subclasses it, and is on its own abstract)?